addons/binding/org.openhab.binding.seneye/src/main/java/org/openhab/binding/seneye/handler/SeneyeHandler.java

```diff
@@ -0,0 +1,135 @@
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.seneye.service.InvalidConfigurationException;
+import org.openhab.binding.seneye.service.ReadingsUpdate;
+import org.openhab.binding.seneye.service.SeneyeDeviceReading;
+import org.openhab.binding.seneye.service.SeneyeService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SeneyeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Niko Tanghe - Initial contribution
+ */
+public class SeneyeHandler extends BaseThingHandler implements ReadingsUpdate {
+
+    private Logger logger = LoggerFactory.getLogger(SeneyeHandler.class);
+    private SeneyeService seneyeService;
+
+    public SeneyeHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (seneyeService == null || seneyeService.isInitialized() == false) {
+            return;
+        }
+
+        if (command instanceof RefreshType) {
+            try {
+                SeneyeDeviceReading readings = seneyeService.getDeviceReadings();
+                newState(readings);
+            } catch (InvalidConfigurationException invalidConfigurationException) {
+                updateStatus(ThingStatus.OFFLINE);
+            }
+        } else {
+            logger.debug("Command {} is not supported for channel: {}", command, channelUID.getId());
+        }
+    }
+
+    @Override
+    public void newState(SeneyeDeviceReading readings) {
+        updateState(CHANNEL_TEMPERATURE, new DecimalType(readings.temperature.curr));
+        updateState(CHANNEL_NH3, new DecimalType(readings.nh3.curr));
+        updateState(CHANNEL_NH4, new DecimalType(readings.nh4.curr));
+        updateState(CHANNEL_O2, new DecimalType(readings.o2.curr));
+        updateState(CHANNEL_PAR, new DecimalType(readings.par.curr));
+        updateState(CHANNEL_PH, new DecimalType(readings.ph.curr));
+        updateState(CHANNEL_LUX, new DecimalType(readings.lux.curr));
+        updateState(CHANNEL_KELVIN, new DecimalType(readings.kelvin.curr));
+        updateState(CHANNEL_LASTREADING, new DateTimeType(readings.status.getLast_experimentDate()));
+        updateState(CHANNEL_SLIDEEXPIRES, new DateTimeType(readings.status.getSlide_expiresDate()));
+    }
+
+    @Override
+    public void invalidConfig() {
+        updateStatus(ThingStatus.OFFLINE);
+    }
+
+    @Override
+    public void initialize() {
+        Configuration conf = this.getConfig();

```

Comment:
No need for `this` here

Reply:
rewritten to getConfigAs(), without this.